Automattic/wp-calypso
# P2: customize signup page

#### Changes proposed in this Pull Request

* Customizes the signup page for P2 flows, as described in p9lV3a-2Lp-p2. This is part of the P2 Onboarding project.
* Refactor: move parts of the customized styles for `accept-invite` signup to `logged-out-form` so it can be reused by the `start/p2` flow

#### Testing instructions 



* Regression test: verify that non-P2 signup flows have the regular signup screen.
* Regression test: verify that accepting invites for non-P2 sites have the regular signup screen.
* Check that nothing looks broken for the following:
    * sign up screen when accepting P2 invites
    * sign up screen for the `/start/p2` flow 
* Check that you can successfully sign up via the `/start/p2` flow 

### Screens
1) Before expanding the email signup form

<img width="800" alt="Screen Shot 2021-08-26 at 11 18 15 PM" src="https://user-images.githubusercontent.com/730823/130990142-a412b8d2-4389-4600-81b7-ac11c231ccf1.png">


2) After expanding the email signup form (at 80%)

<img width="800" alt="Screen Shot 2021-08-26 at 11 18 27 PM" src="https://user-images.githubusercontent.com/730823/130990164-28177788-f8a1-4643-a32a-6261c4d9fd9a.png">

3) (Refactored so styles for email signup form are shared) The `accept-invite-logged-out` screen

<img width="800" alt="Screen Shot 2021-08-26 at 10 21 33 PM" src="https://user-images.githubusercontent.com/730823/130980160-22e13085-bf1c-4d23-b716-127513586293.png">
